### PR TITLE
Feat pyproject migration

### PR DESCRIPTION
**Pull request by sphamba**
# Prework

- [x] I understand and agree to this repository's [code of conduct](https://github.com/ltelab/disdrodb/blob/main/CODE_OF_CONDUCT.md).
- [x] I understand and agree to this repository's [contributing guidelines](https://github.com/ltelab/disdrodb/blob/main/CONTRIBUTING.md).
- [ ] ~I have already submitted an [issue](https://github.com/ltelab/disdrodb/issues) or [discussion thread](https://github.com/ltelab/disdrodb/discussions) to discuss my idea with the maintainers.~

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and communicate accordingly:

**The PR fulfills these requirements:**

- [ ] It's submitted to the branch named as follow : 
  - Add a reader: `reader-<institute>-<campaign>`
  - Fix a bug: `bugfix-<some_key>-<word>`
  - Improve the doc: `doc-<some_key>-<word>`
  - Add a new feature: `feature-<some_key>-<word>`
  - Refactor some code: `refactor-<some_key>-<word>`
  - Optimize some code: `optimize-<some_key>-<word>`
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] Don't forget to link PR to issue if you are solving one.
- [x] All tests are passing.
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

# Related GitHub issues and pull requests

_None_

# Summary

Use `pyproject.toml` in place of `setup.py` and `setup.cfg` to set packaging configuration. Build is performed using
```
pip install build
python -m build
```
instead of `python setup.py ...` (change reflected in Github publishing action).
Automatic version numering with `setuptools-scm`.

## Notes
Support of `setuptools` configuration in `pyproject.toml` is still in [beta](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html). Future versions may introduce breaking changes.
By default, `setuptools-scm` adds all version controled files (i.e. eveything on the Github repo) in the python package. Unwanted files and directories are specified in `MANIFEST.in`.
Configuration for development tools (black, flake8, ruff, ...) may be added to the `pyproject.toml`.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,6 @@
+prune .github
+prune data
+prune docs
+prune disdrodb/L0/manuals
```
Comment (ghiggi) [inline MANIFEST.in:4]:
```suggestion
prune disdrodb/l0/manuals
```

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+prune disdrodb/L0/scripts
```
Comment (ghiggi) [inline MANIFEST.in:5]:
```suggestion
prune disdrodb/l0/scripts
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	"Programming Language :: Python :: 3",
+	"Operating System :: Unix",
+	"Operating System :: MacOS :: MacOS X",
+	"Operating System :: Microsoft :: Windows",
+]
+requires-python = ">=3.10"
+dependencies = [
+	"click",
+	"h5py",
+	"netCDF4",
```
Comment (ghiggi) [inline pyproject.toml:25]:
```suggestion
	"netCDF4",
	"trollsift",
```

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+[tool.setuptools_scm]
+write_to = "disdrodb/_version.py"
+
+[tool.setuptools.packages.find]
+where = ["disdrodb"]
+
+[project.scripts]
+# L0A
+run_disdrodb_l0a_station="disdrodb.L0.scripts.run_disdrodb_l0a_station:run_disdrodb_l0a_station"
```
Comment (ghiggi) [inline pyproject.toml:42]:
```suggestion
run_disdrodb_l0a_station="disdrodb.l0.scripts.run_disdrodb_l0a_station:run_disdrodb_l0a_station"
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+run_disdrodb_l0a="disdrodb.L0.scripts.run_disdrodb_l0a:run_disdrodb_l0a"
```
Comment (ghiggi) [inline pyproject.toml:43]:
```suggestion
run_disdrodb_l0a="disdrodb.l0.scripts.run_disdrodb_l0a:run_disdrodb_l0a"
```

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+# L0B
+run_disdrodb_l0b_station="disdrodb.L0.scripts.run_disdrodb_l0b_station:run_disdrodb_l0b_station"
```
Comment (ghiggi) [inline pyproject.toml:45]:
```suggestion
run_disdrodb_l0b_station="disdrodb.l0.scripts.run_disdrodb_l0b_station:run_disdrodb_l0b_station"
```

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+run_disdrodb_l0_station="disdrodb.L0.scripts.run_disdrodb_l0_station:run_disdrodb_l0_station"
```
Comment (ghiggi) [inline pyproject.toml:46]:
```suggestion
run_disdrodb_l0_station="disdrodb.l0.scripts.run_disdrodb_l0_station:run_disdrodb_l0_station"
```

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+# L0B concatenation
+run_disdrodb_l0b_concat_station="disdrodb.L0.scripts.run_disdrodb_l0b_concat_station:run_disdrodb_l0b_concat_station"
```
Comment (ghiggi) [inline pyproject.toml:48]:
```suggestion
run_disdrodb_l0b_concat_station="disdrodb.l0.scripts.run_disdrodb_l0b_concat_station:run_disdrodb_l0b_concat_station"
```

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+run_disdrodb_l0b_concat="disdrodb.L0.scripts.run_disdrodb_l0b_concat:run_disdrodb_l0b_concat"
```
Comment (ghiggi) [inline pyproject.toml:49]:
```suggestion
run_disdrodb_l0b_concat="disdrodb.l0.scripts.run_disdrodb_l0b_concat:run_disdrodb_l0b_concat"
```

File: pyproject.toml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+# L0
+run_disdrodb_l0b="disdrodb.L0.scripts.run_disdrodb_l0b:run_disdrodb_l0b"
```
Comment (ghiggi) [inline pyproject.toml:51]:
```suggestion
run_disdrodb_l0b="disdrodb.l0.scripts.run_disdrodb_l0b:run_disdrodb_l0b"
```

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+run_disdrodb_l0="disdrodb.L0.scripts.run_disdrodb_l0:run_disdrodb_l0"
```
Comment (ghiggi) [inline pyproject.toml:52]:
```suggestion
run_disdrodb_l0="disdrodb.l0.scripts.run_disdrodb_l0:run_disdrodb_l0"
```

**Review by ghiggi — COMMENTED** [review]
Some changes to be commited to adapt to the last merged PR

**Comment by ghiggi** [comment]
Thanks for having started this. 

I committed some code to take into account the changes I introduced in my last PR.
Before this PR is merged, could you please: 
- resolve the remaining conflict
- Adapt the code in https://github.com/ltelab/disdrodb/blob/main/.github/workflows/release_to_pypi.yml#L29?
 for the automatic release to pypi? With this PR the workflow is broken !

Final questions, where is the current version specified?  
- In _version.py? I don't see it in the repo. 
- Where the version format is specified? 
- If you know how to do it, can you add the `__version__` to the `disdrodb.__init__` file so that one can get the disdrodb 
version by doing: 

```
import disdrodb
disdrodb.__version__
```
 

Thanks in advance 

GG

**Comment by sphamba** [comment]
Hello Gionata,

I addressed your comments:
1. setup.py should not be required anymore, as `python -m build` directly reads `pyproject.toml`, unless you see another reason to keep it?
2. Regis told me that you wanted Python 3.10, but I rolled back to 3.8 without issues (updated `pyproject`)
3. We should test the workflow with this branch merged. Currently, the `release_to_pyp` from the `main` branch is still in use.
4. Done!

Best

**Comment by ghiggi** [comment]
Thanks for the work !
I made a package release and everything seems to work fine. 
Tomorrow I will test it more in-depth.

**Comment by ghiggi** [comment]
@sphamba !
After having made a release to pypi, I tried to install disdrodb with `pip install disdrodb` in a fresh environment.
The requirements are installed correctly without problems, but the package itself got not installed ! 
Can you open another PR and fix this? 
```
>>> import disdrodb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'disdrodb'
```

**Comment by ghiggi** [comment]
@@sphamba I fixed it by replacing:
```
[tool.setuptools.packages.find]
where = ["disdrodb"]
```
 with
```
[tool.setuptools]
license-files = ["LICENSE"]
packages = ["disdrodb"]  
```

If you use [tool.setuptools.packages.find] then you should also manually specify `include` and `exclude` arguments !

**Pull request merged**